unstructured/metrics/evaluate.py

```diff
@@ -59,13 +60,23 @@ def measure_text_extraction_accuracy(
     if not output_list:
         print("No output files to calculate to edit distances for, exiting")
         sys.exit(0)
+    if output_type not in ["json", "txt"]:
+        raise ValueError(
+            f"Specified file type under `output_dir` or `output_list` should be one of \
+                'json' or 'txt'. The given file type is {output_type}, exiting."
+        )
+    if not all(_.endswith(output_type) for _ in output_list):
+        print(
+            "The directory contains file type inconsistent with the given input. \
+                Please note that some files will be skipped."
+        )
 
     rows = []
 
     # assumption: output file name convention is name-of-file.doc.json
     # NOTE(klaijan) - disable=True means to not show, disable=False means to show the progress bar
     for doc in tqdm(output_list, leave=False, disable=not visualize):  # type: ignore
-        filename = (doc.split("/")[-1]).split(".json")[0]
+        filename = (doc.split("/")[-1]).split(f".{output_type}")[0]
```

Comment:
a small trick: here using split to just drop the file extension actually lead to creation of a list (i.e., expensive)
here we really need is the part of the string before the extension so we can do something like 
```
ext_index = -len(output_type)
for doc in tqdm(...):
    filename = os.path.basename(doc)[:ext_index]
```
also here using split on "/" is potentially problematic as it assumes the os separator is "/"